code/game/area/areas.dm

```diff
@@ -490,3 +497,130 @@ GLOBAL_LIST_EMPTY(teleportlocs)
 /// Called when a living mob that spawned here, joining the round, receives the player client.
 /area/proc/on_joining_game(mob/living/boarder)
 	return
+
+// Day night stuff
+/**
+ * This will calculate all adjacent turfs in this area and add them to a cache for lighting effects.
+ *
+ * WARNING: This proc is VERY expensive and should be used sparingly.
+ */
+/area/proc/initialize_day_night_adjacent_turfs()
+	LAZYCLEARLIST(adjacent_day_night_turf_cache)
+	LAZYINITLIST(adjacent_day_night_turf_cache)
+
+	for(var/turf/iterating_turf in contents)
```

Comment:
i bet a large part of the cost of this proc is iterating area contents right here. secretly because byond is determined to be dumb, iterating area contents actually loops through every turf in the world, skipping turfs that arent inside that area. youll likely get better results by iterating all the turfs in a z level and updating each ones area there

Reply:
good shout, I will look into this shortly

Reply:
Though this proc is called once during startup, and changes only when a shuttle departs